omx/gstomxvideofilter.c

```diff
@@ -362,6 +362,8 @@ gst_omx_video_filter_init (GstOMXVideoFilter * self,
   self->output_buffers = GST_OMX_VIDEO_FILTER_OUTPUT_BUFFERS_DEFAULT;
 
   g_rec_mutex_init (&self->stream_lock);
+  g_mutex_init (&self->filter_lock);
```

Comment:
Let the reader know (briefly) why this is used. Something like: _The lock helps processing 1 buffer at a time, which in consequence prevents pipe deadlock during reverse playback_

Reply:
Comments added in b5747cd1914ada2395121f1cb91323dc2b665e84